### ISSUE TITLE
macOS 11.0 Big Sur

### ISSUE THREAD

**Issue by Mattrak** [justdan96/tsMuxer#373]
The app won't open at all, and stuck forever at start

**Comment by glc650** [comment]
Same here.

**Comment by justdan96** [comment]
I don't have a copy of Big Sur to test with, can you please try to open the command line version of tsMuxer? You need to extract the .app file, maybe Keka can do it? https://www.keka.io/en/

**Comment by glc650** [comment]
You are talking about the "tsMuxeR" file inside of tsMuxerGUI.app?  You can get to that and the rest of the contents of tsMuxerGUI.app by right clicking and selecting Show Contents.  Where do I extract it to and what flags do I run with it?

**Comment by glc650** [comment]
sudo /Users/garrett/Downloads/tsMuxeR --version
tsMuxeR version git-5e7ca8a. github.com/justdan96/tsMuxer

**Comment by justdan96** [comment]
Can you please try to run tsMuxerGUI from the command line as well?

**Comment by moderndeveloperllc** [comment]
@justdan96 I can the "native" build running fine i.e., the artifact found here: https://github.com/justdan96/tsMuxer/actions/runs/346066234. @glc650 Can you try that one and see? It's better as it recognizes high DPI screens, and also light/dark mode.

The tabs along the top aren't the newer style, but that's just visual.
<img width="462" alt="ScreenShot 2020-11-16 at 8 21 20 AM" src="https://user-images.githubusercontent.com/1920405/99263404-bfdeab80-27e4-11eb-8d2c-4fb402725e98.png">

**Comment by glc650** [comment]
> @justdan96 I can the "native" build running fine i.e., the artifact found here: https://github.com/justdan96/tsMuxer/actions/runs/346066234. @glc650 Can you try that one and see? It's better as it recognizes high DPI screens, and also light/dark mode.
> 
> The tabs along the top aren't the newer style, but that's just visual.
> <img alt="ScreenShot 2020-11-16 at 8 21 20 AM" width="462" src="https://user-images.githubusercontent.com/1920405/99263404-bfdeab80-27e4-11eb-8d2c-4fb402725e98.png">
That one loads for me (haven't actually tried using it yet though)

**Comment by Mattrak** [comment]
Yes that one works for me too

**Comment by glc650** [comment]
Just saw the 11/17 release and tried it but it's still broken.  So far only the release above works with Big Sur.

**Comment by Mattrak** [comment]
This solution suits me

**Comment by moderndeveloperllc** [comment]
@justdan96 Not sure this should have been closed if the most readily available file is not working.

**Comment by justdan96** [comment]
We use [this](https://github.com/justdan96/tsmuxer_build) Docker image to build the nightlies in Bintray, which uses osxcross to generate the OSX builds. To help us diagnose the issue can you try running the GUI with `strace`? 

Meanwhile I am trying to set up a Big Sur VM today, so hopefully I can try to reproduce the issue myself.

**Comment by justdan96** [comment]
I have reproduced the issue when using the releases from Bintray. The same issue does not appear with the Mac native version. Even with the Mac native version the GUI looks a little odd - I think it is due to this issue: https://bugreports.qt.io/browse/QTBUG-85546. I can try updating Qt in the Docker container but it is actually slightly newer than the Mac native one is being built on.

It might be because we are targeting MacOS 10.10, if we try targeting MacOS 10.13 instead maybe we will get better luck. We may have to ensure we are using a later version of Xcode. 

Long story short - it may take us some time to support Big Sur with the Bintray releases, as we have to do quite a bit of investigaton and updating of the Docker container.

**Comment by moderndeveloperllc** [comment]
@justdan96 Thanks for looking into this. Only about [10% of the macOS users worldwide](https://gs.statcounter.com/macos-version-market-share/desktop/worldwide) are using something less than 10.13 and Apple is no longer doing security fixes for less than 10.13 - basically anything lower is EOL. Honestly, if I were in your shoes I would put the time into seeing if you can, as a finishing step in the native build, upload the native version to Bintray and overwrite the osxcross version currently being uploaded. Not an expert on Github Actions though, so that might not be maintainable.

**Comment by justdan96** [comment]
I said 10.13 as that is still officially supported, we can move up through the versions as they go out of official support. I'm reticent to switch to the MacOS native build officially as the idea was that we can use one Docker container to produce all of our Windows/Mac/Linux releases. This allows offline creation of builds as well, so we aren't solely dependant on GitHub Actions.

Switching to native builds is certainly something we will have to think about but seeing as Big Sur was only released 9 days ago and none of the developers are Mac users you can understand that it could take us some time to get a full resolution to the issue.

**Comment by xavery** [comment]
Alright, the first Mac native nightly build is now available : https://bintray.com/justdan96/tsMuxer/tsMuxerGUI-Nightly/2020-12-05#files . Unfortunately, the rest is missing due to me screwing up and forgetting to make `bintray_nightly_check.sh` executable, which resulted in the "docker" part of the build failing too early.
Things should be back to normal with 2020-12-06, which will be triggered due to a new commit today.

**Comment by xavery** [comment]
Alright, tonight's build has just finished successfully. I think this can be closed once somebody confirms that the Mac build available [at Bintray](https://bintray.com/justdan96/tsMuxer/tsMuxerGUI-Nightly/2020-12-06#files) runs okay on 11.0.

**Comment by glc650** [comment]
It launches fine for me but I haven't tested any further than that.

**Comment by justdan96** [comment]
I will be soon pushing an update to the Docker container, I've made some changes so the produced executables will be compatible with MacOS 13.0-16.0.

@xavery anything else I should update in the Docker container while I'm at it?

**Comment by xavery** [comment]
@justdan96 Not really. The question is, do you want to go back to Docker-based nightly builds for the Mac platform?

**Comment by justdan96** [comment]
The Mac builds we have have been working for a while, closing this one.
